data/backstage/infra/2025-10-30-how-the-opam-repository-works.md

```diff
@@ -0,0 +1,116 @@
+
+At the heart of the repository's reliability is an extensive continuous integration (CI) system built around [OCurrent](https://ocurrent.org), which powers the [opam-repo-ci service](https://opam.ci.ocaml.org). When you submit a pull request to add or update a package, this system:
+
+- **Validates package metadata** using [an extensive set of linting rules](https://github.com/ocurrent/opam-repo-ci/blob/babb8b03e5eabf8e5ecc12384587a8303004a22b/opam-ci-check/lint/lint_error.ml#L13-L37)
+- **Tests compilation** across multiple OCaml compiler versions (currently supporting back to OCaml 4.08)
+- **Verifies installation** on various platforms including Linux distributions, macOS, FreeBSD, Windows, and multiple CPU architectures (x86_64, ARM64, ARM32, PowerPC, s390x, RISC-V)
+- **Runs package tests** to ensure functionality
+- **Checks reverse dependencies** to ensure your changes don't break existing packages that depend on your package
+
+This comprehensive testing matrix helps maintain the repository's high quality standards. Windows support was added in 2024, marking a significant expansion of the platform coverage.
+
+### Supporting Infrastructure
+
+The opam repository integrates with several other OCaml infrastructure services:
+
+- **Docker Base Images**: The [docker-base-images service](https://images.ci.ocaml.org) provides consistent testing environments, building official images for various Linux distributions, OCaml versions, and architectures
+- **Documentation CI**: The [docs.ci.ocaml.org service](https://docs.ci.ocaml.org) automatically builds and publishes package documentation to the [OCaml.org website's package area](https://ocaml.org/packages)
+- **Health Check Services**: Multiple services continuously monitor package installability across different platforms and compiler versions:
+  - [check.ci.ocaml.org](https://check.ci.ocaml.org): Tests all packages regularly
+  - [dune.check.ci.dev](https://dune.check.ci.dev): Tests packages with the Dune build system - related to the recent addition of the Dune Package Management feature
+  - [windows.check.ci.dev](https://windows.check.ci.dev): Windows-specific testing
+  - [freebsd.check.ci.dev](https://freebsd.check.ci.dev): FreeBSD compatibility
+  - [oxcaml.check.ci.dev](https://oxcaml.check.ci.dev): Tests with OxCaml compiler variant
+
+#### Infrastructure Funding
+
+As of 2025, the OCaml infrastructure is primarily funded and supported by the Cambridge Computer Lab and Tarides. Additional cloud hosting costs are covered by community contributions. For architecture-specific testing, [Oregon State University Open Source Lab](https://osuosl.org/communities/) provides essential PowerPC infrastructure support, and Jane Street has generously donated x86 machines to Cambridge University. This combination of institutional support, corporate sponsorship, and community backing enables the comprehensive multi-platform testing that ensures package quality across the entire OCaml ecosystem.
```

Comment:
```suggestion
As of 2025, the OCaml.org infrastructure is primarily funded and supported by the Cambridge Computer Lab and Tarides. Additional cloud hosting costs are covered by community contributions. For architecture-specific testing, [Oregon State University Open Source Lab](https://osuosl.org/communities/) provides essential PowerPC infrastructure support, and Jane Street has generously donated x86 machines to Cambridge University. This combination of institutional support, corporate sponsorship, and community backing enables the comprehensive multi-platform testing that ensures package quality across the entire OCaml ecosystem.
```

just a minor tweak: the ocaml.org infrastructure. Core OCaml compiler infra is run by Inria (the compiler Jenkins)

Reply:
Yes, compiler folk -- don't mention Jenkins on my account. My point was that we call out support from OSUOSL but not the IBM/Marist teams for the S390 support.